openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -53,7 +53,7 @@ def get_ctas(self, xblock, category, completed):
             if self._is_block_shiftable(xblock, category):
                 ctas.append(self._make_reset_deadlines_cta(xblock, category, is_learning_mfe))
 
-        elif category == self.VERTICAL_BANNER and not completed:
+        elif category == self.VERTICAL_BANNER and not completed and missed_deadlines:
```

Comment:
Does this mean the button always shows? Like, first time I roll up to an assignment, I'm seeing "shift date" buttons?

Reply:
No. The criteria to show at the problem level are
- self-paced
- able to attempt
- and past-due
Assuming they roll up to the assignment prior to the due date, they won't see anything. If the first time is after the due date, then yes, it will show, but it needs to because the submit button will be disabled.

Reply:
OK we talked a bit more about this on slack - it's intentional that we aren't going to be doing the other missed-deadline logic checks (like is-the-problem-scored) and that we don't want the banner in these cases. Let's try to remember this distinction when we write fuller docs for PLS.

Reply:
For clarity, ungraded content will never have a due date in PLS so the shift dates logic will never apply for those problems (they are always submittable). And if a user has missed_gated_content, the function will still early exit and not allow shifting of dates.